src/pybind/mgr/dashboard_v2/CMakeLists.txt

```diff
@@ -9,23 +9,33 @@ add_custom_target(mgr-dashboard_v2-test-venv
 add_dependencies(tests mgr-dashboard_v2-test-venv)
 
 if(WITH_MGR_DASHBOARD_V2_FRONTEND AND NOT CMAKE_SYSTEM_PROCESSOR MATCHES "aarch64|AARCH64|arm|ARM")
-  find_program(NPM_BIN
-    NAMES npm
-    HINTS $ENV{NPM_ROOT}/bin)
-  if(NOT NPM_BIN)
-    message(FATAL_ERROR "WITH_MGR_DASHBOARD_V2_FRONTEND set, but npm not found")
-  endif()
+
+set(mgr-dashboard_v2-nodeenv ${CMAKE_SOURCE_DIR}/build/dashboard_nodeenv)
```

Comment:
should use `CMAKE_BINARY_DIR` instead of hardwiring it to `${CMAKE_SOURCE_DIR}/build`. also i'd suggest remove the prefix of `mgr-dashboard_v2-`, and install the nodeenv in current binary, i.e.

```cmake
set(mgr-dashboard_v2-nodeenv ${CMAKE_CURRENT_BINARY_DIR}/node-env)
```